### ISSUE TITLE
Performance improvements for rule set conflict fixing

### ISSUE THREAD

**Issue by fmallet** [SonarSource/sonarlint-visualstudio#187]
We analyze and fix the issues on the UI thread for simplicity.
In most cases this is not an performance issue since that time it takes to do both is less than 100ms. In some cases (i.e. roslyn) this could go up to approx. 3sec (measured on prototype, and could be different on final bits).
We can do various things (all should be possible, minor tweeaks might be required, more on the test side):
1. Analyze and fix in parallel
2. Separate the fix into two stages - in-memory (background thread) and persisting (ui-thread).
Note: we already aggregate the rulesets based on project, so if same rulesets were used for more than one configuration, they will be analyzed once and fixed once.

Comes from [SVS-67](https://jira.sonarsource.com/browse/SVS-67)

**Comment by Evangelink** [comment]
No longer relevant for the new connected mode.
